htdocs/core/tpl/extrafields_list_search_sql.tpl.php

```diff
@@ -35,7 +35,8 @@
 		{
 			$mode_search = 0;
 			if (in_array($typ, array('int', 'double', 'real', 'price'))) $mode_search = 1; // Search on a numeric
-			if (in_array($typ, array('sellist', 'link')) && $crit != '0' && $crit != '-1') $mode_search = 2; // Search on a foreign key int
+			if (in_array($typ, array('sellist', 'link')) && $crit != '0' && $crit != '-1' && !is_string($crit)) $mode_search = 2; // Search on a foreign key int
```

Comment:
!is_string is dangerous because 2 i snot string but '2' is.
Can't you use instead is_numeric()  ?

Reply:
Damly right, fix